### PR TITLE
Add metrics to track controller segment download and upload requests in progress

### PR DESCRIPTION
**Pull request by gviedma**
This change tracks the number of in progress segment download and upload requests currently being processed by the controller via their respective Gauge metrics _segmentDownloadsInProgress_ and _segmentUploadsInProgress_.  These two metrics are useful to track the number of in-flight operations in large clusters, where a large number of concurrent uploads/downloads may bring down a shared controller.

This change also introduces a generic JAX-RS annotation-based mechanism that can be reused to track in-flight request metrics of other types in the future. See _InflightRequestMetricsInterceptor_.

### CODE REVIEW

**Review by Jackie-Jiang — APPROVED** [review]
LGTM!

File: pinot-common/src/main/java/org/apache/pinot/common/metrics/ControllerGauge.java [hunk 1]
```diff
@@ -110,7 +110,14 @@ public enum ControllerGauge implements AbstractMetrics.Gauge {
   MISSING_CONSUMING_SEGMENT_NEW_PARTITION_COUNT("missingConsumingSegmentNewPartitionCount", false),
 
   // Maximum duration of a missing consuming segment in ideal state (in minutes)
-  MISSING_CONSUMING_SEGMENT_MAX_DURATION_MINUTES("missingSegmentsMaxDurationInMinutes", false);
+  MISSING_CONSUMING_SEGMENT_MAX_DURATION_MINUTES("missingSegmentsMaxDurationInMinutes", false),
+
+  // Number of in progress segment downloads
+  CONTROLLER_SEGMENT_DOWNLOADS_IN_PROGRESS_COUNT("segmentDownloadsInProgressCount", true),
```
Comment (mcvsubbu) [inline pinot-common/src/main/java/org/apache/pinot/common/metrics/ControllerGauge.java:116]: Maybe remove `CONTROLLER_` prefix? It is in `ControllerGauge`, emitted by the controller, so we are fine to remove the prefix? Also, maybe remove the `_COUNT` suffix also? 

Metric names are getting long....

**Review by mcvsubbu — COMMENTED** [review]
So, we should also add the realtime segment upload to the metric.

Done in the class: `LLCSegmentCompletionHandlers`

This class is invoked by the servers to complete a realtime segment

File: pinot-common/src/main/java/org/apache/pinot/common/metrics/ControllerGauge.java (hunk 1 above)
Comment (gviedma) [reply pinot-common/src/main/java/org/apache/pinot/common/metrics/ControllerGauge.java]: Good point, I've dropped both the `CONTROLLER_` prefix and the `_COUNT` suffix for readability.

**Review by mcvsubbu — APPROVED** [review]

**Pull request merged**